mysql-persistence/src/main/java/com/netflix/conductor/dao/mysql/MySQLExecutionDAO.java

```diff
@@ -0,0 +1,761 @@
+		indexer.add(log);
+	}
+
+	@Override
+	public void removeTask(String taskId) {
+		Task task = getTask(taskId);
+
+		if(task == null) {
+			logger.warn("No such Task by id {}", taskId);
+			return;
+		}
+
+		String taskKey = task.getReferenceTaskName() + "" + task.getRetryCount();
+
+		withTransaction(connection -> {
+			removeScheduledTask(connection, task, taskKey);
+			removeWorkflowToTaskMapping(connection, task);
+			removeTaskInProgress(connection, task);
+			removeTaskData(connection, task);
+		});
+	}
+
+	@Override
+	public Task getTask(String taskId) {
+		String taskJsonStr = getWithTransaction(c -> c.createQuery(GET_TASK).addParameter("taskId", taskId).executeScalar(String.class));
+		return taskJsonStr != null ? readValue(taskJsonStr, Task.class) : null;
+	}
+
+	@Override
+	public List<Task> getTasks(List<String> taskIds) {
+		if (taskIds.isEmpty()) return Lists.newArrayList();
+
+		List<String> taskIdParameters = taskIds.stream().map(taskId -> format("'%s'", taskId)).collect(Collectors.toList());
+
+		return getWithTransaction(c -> c.createQuery(format(GET_TASKS, join(taskIdParameters, ","))).executeScalarList(String.class))
+				.stream()
+				.filter(Objects::nonNull)
+				.map(taskJsonStr -> readValue(taskJsonStr, Task.class))
+				.collect(Collectors.toList());
+	}
+
+	@Override
+	public List<Task> getPendingTasksForTaskType(String taskName) {
+		Preconditions.checkNotNull(taskName, "task name cannot be null");
+
+		List<String> taskIds = getWithTransaction(connection -> connection.createQuery(GET_ALL_IN_PROGRESS_FOR_TASK_TYPE)
+				.addParameter("taskDefName", taskName)
+				.executeScalarList(String.class));
+
+		return getTasks(taskIds);
+	}
+
+	@Override
+	public List<Task> getTasksForWorkflow(String workflowId) {
+		List<String> taskIds = getWithTransaction(c -> c.createQuery(GET_TASKS_FOR_WORKFLOW)
+				.addParameter("workflowId", workflowId)
+				.executeScalarList(String.class));
+
+		return getTasks(taskIds);
+	}
+
+	@Override
+	public String createWorkflow(Workflow workflow) {
+		workflow.setCreateTime(System.currentTimeMillis());
+		return insertOrUpdateWorkflow(workflow, false);
+	}
+
+	@Override
+	public String updateWorkflow(Workflow workflow) {
+		workflow.setUpdateTime(System.currentTimeMillis());
+		return insertOrUpdateWorkflow(workflow, true);
+	}
+
+	@Override
+	public void removeWorkflow(String workflowId) {
+		try {
+			Workflow wf = getWorkflow(workflowId, true);
+
+			//Add to elasticsearch
+			indexer.update(workflowId, new String[]{RAW_JSON_FIELD, ARCHIVED_FIELD}, new Object[]{om.writeValueAsString(wf), true});
+
+			withTransaction(connection -> {
+				removeWorkflowDefToWorkflowMapping(connection, wf);
+				removeWorkflow(connection, workflowId);
+				removePendingWorkflow(connection, wf.getWorkflowType(), workflowId);
+			});
+
+			for(Task task : wf.getTasks()) {
+				removeTask(task.getTaskId());
+			}
+
+		} catch(Exception e) {
+			throw new ApplicationException(e.getMessage(), e);
```

Comment:
No reason to copy the message of the cause. There's probably another string like `"Unable to remove workflow " + workflowId` that could be used here.

Reply:
agree, most catch handlers I have copy-pasted them from the the RedisExecutionDAO and haven't paid much attention

I will review all try/catch blocks to make sure they are consistent